enaml/core/compiler_helpers.py

```diff
@@ -736,6 +739,26 @@ def add_decl_function(node, func, is_override):
     setattr(klass, name, d_func)
 
 
+def wrap_function(func, scope):
+    """Wrap a function to call it with the dynamicscope in which it was defined
+
+    Parameters
+    ----------
+    func : FunctionType
+        Function for which a new scope need to be built
+
+    scope : DynamicScope
+        Scope in which the function was defined.
+
+    """
+    def wrapper(*args, **kwargs):
+        return call_func(func, args, kwargs, scope)
+
+    update_wrapper(wrapper, func)
```

Comment:
why this over `@functools.wraps`?

Reply:
No reason I will use `functools.wraps`

Reply:
Actually it is because wraps expects the decorator to take as single argument the function.